src/EditorFeatures/CSharp/BlockCommentEditing/BlockCommentEditingCommandHandler.cs

```diff
@@ -55,186 +56,199 @@ public bool ExecuteCommand(ReturnKeyCommandArgs args, CommandExecutionContext co
+            var startsWithBlockCommentMiddleString = currentLine.StartsWith(firstNonWhitespacePosition, "*", ignoreCase: false);
 
-            if (!currentLineStartsWithBlockCommentStartString && !currentLineStartsWithBlockCommentMiddleString)
+            if (!startsWithBlockCommentStartString &&
+                !startsWithBlockCommentMiddleString)
             {
                 return null;
             }
 
-            if (!IsCaretInsideBlockCommentSyntax(caretPosition))
-            {
+            if (!IsCaretInsideBlockCommentSyntax(caretPosition, out var document, out var blockComment))
                 return null;
-            }
 
-            if (currentLineStartsWithBlockCommentStartString)
-            {
-                if (BlockCommentEndsRightAfterCaret(caretPosition))
-                {
-                    //      /*|*/
-                    return " ";
-                }
-                else if (caretPosition == firstNonWhitespacePosition + 1)
-                {
-                    //      /|*
-                    return null; // The newline inserted could break the syntax in a way that this handler cannot fix, let's leave it.
-                }
-                else
-                {
-                    //      /*|
-                    return " *" + GetPaddingOrIndentation(currentLine, caretPosition, firstNonWhitespacePosition, "/*");
-                }
-            }
+            var textSnapshot = caretPosition.Snapshot;
 
-            if (currentLineStartsWithBlockCommentEndString)
+            // The whitespace indentation on the line where the block-comment starts.
+            var commentIndentation = textSnapshot.GetText(Span.FromBounds(
+                textSnapshot.GetLineFromPosition(blockComment.FullSpan.Start).Start,
+                blockComment.FullSpan.Start));
+
+            // The whitespace indentation on the current line up to the first non-whitespace char.
+            var lineIndentation = textSnapshot.GetText(Span.FromBounds(
+                currentLine.Start,
+                firstNonWhitespacePosition));
+
+            var exteriorText = GetExteriorText();
+            if (exteriorText == null)
+                return null;
+
+            var options = document.Project.Solution.Options;
+            var newLine = options.GetOption(FormattingOptions.NewLine, LanguageNames.CSharp);
+            return newLine + exteriorText;
+
+            string GetExteriorText()
             {
-                if (BlockCommentEndsRightAfterCaret(caretPosition))
+                if (startsWithBlockCommentStartString)
                 {
-                    //      /*
-                    //      |*/
-                    return " ";
+                    if (BlockCommentEndsRightAfterCaret(caretPosition))
+                    {
+                        //      /*|*/
+                        return commentIndentation + " ";
```

Comment:
i find this much easier to read now as well.  i.e. the indentation of the next line should be the indentation of the comment, plus a single space.

Reply:
hrmm.  comments shouldn't havebeen lost.  that wasn't intentional.  let me review and add back in.